src/python_routing_v01/compute_nhd_routing_SingleSeg.py

```diff
@@ -453,40 +697,60 @@ def main():
         print("reach organization complete")
     if showtiming:
         print("... in %s seconds." % (time.time() - start_time))
-
-    if showtiming:
         start_time = time.time()
+
     connections = supernetwork_values[0]
 
-    flowdepthvel = {
+    # initialize flowveldepth dict
+    flowveldepth = {
         connection: {
-            "flow": {"prev": 0, "curr": 0},
-            "depth": {"prev": 0, "curr": 0},
-            "vel": {"prev": 0, "curr": 0},
-            "qlat": {"prev": 0, "curr": 0},
+            "qlatval": [],
+            "time": [],
+            "flowval": [],
+            "velval": [],
+            "depthval": [],
         }
         for connection in connections
     }
 
-    parallelcompute = False
-    if not parallelcompute:
+    # Lateral flow
+    if (
+        run_pocono_test
+    ):  # test 1. Take lateral flow from wrf-hydro output from Pocono Basin
+        ql_input_folder = os.path.join(
+            root, r"test/input/geo/PoconoSampleData2/Pocono_ql_testsamp1_nwm_mc.csv"
+        )
+        ql = pd.read_csv(ql_input_folder, index_col=0)
+
+    else:
+        q = np.full((len(connections), nts), qlat_const, dtype="float32")
+        ql = pd.DataFrame(q, index=connections.keys(), columns=range(nts))
```

Comment:
Pandas has this use case covered.
```python
ql = pd.DataFrame(qlat_const, index=connections.keys(), columns=range(nts))
```
No need to allocate the numpy array first.

Reply:
fixed per comment.